crates/transpiler/src/passes/litinski_transformation.rs

```diff
@@ -182,29 +280,48 @@ pub fn run_litinski_transformation(
 
                     new_dag.apply_operation_back(
                         py_gate.into(),
-                        &evo_qubits,
-                        &no_clbits,
+                        &indices,
+                        &[],
                         Some(Parameters::Params(smallvec![time])),
                         None,
                         #[cfg(feature = "cache_pygates")]
                         None,
                     )?;
                 }
-                "measure" => {
+                OperationRef::StandardInstruction(StandardInstruction::Measure) => {
                     // Returns the evolved Pauli in the sparse format: (sign, pauli z, pauli x, indices),
                     // where signs `true` and `false` correspond to coefficients `-1` and `+1` respectively.
-                    let (sign, z, x, indices) = clifford.get_inverse_z(qubits[0]);
+                    let (sign, terms, indices) =
+                        clifford.get_inverse_z(dag.get_qargs(inst.qubits)[0].index());
+                    let mut z: Vec<bool> = Vec::with_capacity(terms.len());
+                    let x: Vec<bool> = terms
+                        .into_iter()
```

Comment:
Now `z` and `x` are treated a bit differently. Do you think that this is faster than also preallocating `x` with the given capacity and calling `terms.iter().for_each() { ... }`?

Reply:
I did this in: https://github.com/Qiskit/qiskit/pull/15315/commits/a76941bf32307e108a5400905726402dbd76776b